tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py

```diff
@@ -203,12 +203,67 @@ def test_automated_recovery_from_failed_nodes_IPI_reactive(
 @ignore_leftovers
 @tier4
 @tier4a
-@aws_based_platform_required
 @ipi_deployment_required
 class TestAutomatedRecoveryFromStoppedNodes(ManageTest):
 
     osd_worker_node = None
     extra_node = False
+    machineset_name = None
+    start_ready_replica_count = None
+
+    def change_osd_worker_node_to_ready_state(self, nodes):
+        """
+        Change the osd worker node to be in Ready state.
+        """
+        expected_statuses = [constants.NODE_NOT_READY, constants.NODE_READY]
+        node_status = get_node_status(self.osd_worker_node[0])
+        node_name = self.osd_worker_node[0].name
+
+        log.info(f"The status of the node {node_name} is {node_status} ")
+
+        if node_status not in expected_statuses:
+            log.warning(
+                f"The node {node_name} is not in the expected statuses: {expected_statuses}. "
+                f"Trying to restart the node..."
+            )
+            nodes.restart_nodes_by_stop_and_start(
+                nodes=self.osd_worker_node, force=True
+            )
+            return
+
+        if node_status == constants.NODE_NOT_READY:
+            log.info(f"Starting the node {node_name}...")
+            nodes.start_nodes(nodes=self.osd_worker_node, wait=True)
+            log.info(f"Successfully started node {node_name} instance")
+        else:
+            log.info(
+                f"The node {node_name} is already in the expected status {constants.NODE_READY}"
+            )
+
+    def wait_for_current_ready_replica_count_equal_to_start_ready_replica_count(self):
+        """
+        Wait for the current ready replica count to be equal to the ready replica count
+        at the beginning of the test.
+        """
+        current_ready_replica_count = machine.get_ready_replica_count(
+            self.machineset_name
+        )
+        log.info(
+            f"current ready replica count = {current_ready_replica_count}, "
+            f"start ready replica count = {self.start_ready_replica_count}"
+        )
+        timeout = 180
+        log.info(
+            f"Wait {timeout} seconds for the current ready replica count to be equal "
+            f"to the start ready replica count"
+        )
+        sample = TimeoutSampler(
+            timeout=timeout,
+            sleep=10,
+            func=machine.get_ready_replica_count,
+            machine_set=self.machineset_name,
+        )
+        sample.wait_for_func_value(value=self.start_ready_replica_count)
```

Comment:
These functions should not be inside the test file. Please take them our to a relevant utils or helpers file

Reply:
Yes, that will do.